unblob/handlers/archive/instar/instar_hd.py

```diff
@@ -0,0 +1,44 @@
+from pathlib import Path
+
+from structlog import get_logger
+
+from unblob.file_utils import File, iterate_file
+from unblob.handlers.archive.zip import ZIPHandler
+from unblob.models import Extractor, HexString
+
+logger = get_logger()
+
+LOCAL_FILE_HEADER_ZIP = b"\x50\x4b\x03\x04"
+LOCAL_FILE_HEADER_INSTAR = b"\x50\x4b\x03\x07"
+
+EOCD_ZIP = b"\x50\x4b\x05\x06"
+EOCD_INSTAR = b"\x50\x4b\x05\x09"
+
+CD_FILE_HEADER_ZIP = b"\x50\x4b\x01\x02"
+CD_FILE_HEADER_INSTAR = b"\x50\x4b\x01\x08"
+
+
+class InstarHDExtractor(Extractor):
+    def fix_headers(self, chunk: bytes) -> bytes:
+        return (
+            chunk.replace(LOCAL_FILE_HEADER_INSTAR, LOCAL_FILE_HEADER_ZIP)
+            .replace(EOCD_INSTAR, EOCD_ZIP)
+            .replace(CD_FILE_HEADER_INSTAR, CD_FILE_HEADER_ZIP)
+        )
+
+    def extract(self, inpath: Path, outdir: Path):
+        with File.from_path(inpath) as file:
+            outpath = outdir / "instar_hd.deobfuscated"
+            with outpath.open("wb") as outfile:
+                for chunk in iterate_file(file, 0, inpath.stat().st_size):
+                    outfile.write(self.fix_headers(chunk))
```

Comment:
What about 
- copying the file to `instar_hd.deobfuscated`
- `mmap` the new file
- fix headers on the mmap

this would leave out the `iterate_file` which could cause an extraction failure, as mentioned in https://github.com/onekey-sec/unblob/pull/537#issuecomment-1503165259